src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -139,7 +139,16 @@ HostSymbolStartAnalysisScope getSymbolAnalysisScopeCore()
                             lock (_gate)
                             {
                                 _lazyPendingMemberSymbolsMapOpt = _lazyPendingMemberSymbolsMapOpt ?? new Dictionary<ISymbol, HashSet<ISymbol>>();
-                                _lazyPendingMemberSymbolsMapOpt.Add(symbol, dependentSymbols);
+
+                                // Guard against entry added from another thread.
+                                if (!_lazyPendingMemberSymbolsMapOpt.ContainsKey(symbol))
```

Comment:
_lazyPendingMemberSymbolsMapOpt[symbol] = dependentSymbols? if you don't care about debug asset.